lib/std.rs

```diff
@@ -145,6 +145,62 @@ impl<T> From<std_Option<T>> for Option<T> {
     }
 }
 
+impl<A: record::FromRecord + serde::de::DeserializeOwned + Default> record::FromRecord
+    for std_Option<A>
+{
+    fn from_record(val: &record::Record) -> result::Result<Self, String> {
+        match val {
+            record::Record::PosStruct(constr, args) => match constr.as_ref() {
+                "std.None" if args.len() == 0 => Ok(std_Option::std_None {}),
+                "std.Some" if args.len() == 1 => Ok(std_Option::std_Some {
+                    x: <A>::from_record(&args[0])?,
+                }),
+                c => result::Result::Err(format!(
+                    "unknown constructor {} of type std_Option in {:?}",
+                    c, *val
+                )),
+            },
+            record::Record::NamedStruct(constr, args) => match constr.as_ref() {
+                "std.None" => Ok(std_Option::std_None {}),
+                "std.Some" => Ok(std_Option::std_Some {
+                    x: record::arg_extract::<A>(args, "x")?,
+                }),
+                c => result::Result::Err(format!(
+                    "unknown constructor {} of type std_Option in {:?}",
+                    c, *val
+                )),
+            },
+            /* `Option` encoded as an array of size 0 or 1.  This is, for instance, useful when
+             * interfacing with OVSDB. */
+            record::Record::Array(kind, records) => match (records.len()) {
```

Comment:
what about other collections?

Reply:
sets, maps, etc., are extern types and already have their own manual implementations of `FromRecord` that support arrays.